libraries/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -273,14 +270,15 @@ static bool _topicFilterMatch( const char * pTopicName,
         filterIndex++;
     }
 
-    /* If the end of both strings has been reached, they match. */
-    if( ( nameIndex == topicNameLength ) && ( filterIndex == topicFilterLength ) )
+    if( status == false )
     {
-        status = true;
+        /* If the end of both strings has been reached, they match. */
+        if( ( nameIndex == topicNameLength ) && ( filterIndex == topicFilterLength ) )
+        {
+            status = true;
+        }
```

Comment:
Nit: I like reducing the number of `if`s whenever possible, and it looks like `status = ( nameIndex == topicNameLength ) && ( filterIndex == topicFilterLength );` would be equivalent to this block. Is there a disadvantage to this?

Reply:
Boolean expression in C returns 0 or 1. It does not return bool datatype.  I believe that's the reason we don't use boolDataType = expression.

Reply:
Since it is used elsewhere I'll make the change.